src/simplify.jl

```diff
@@ -166,7 +166,18 @@ end
 
 ### Simplification rules
 
-simplify(x, rules=SIMPLIFY_RULES) = RuleSet(rules)(x)
+to_symbolic(x::Symbolic) = x
+function to_symbolic(x)
+    if !istree(x)
+        return x
+    end
+
+    Term(operation(x),
+         symtype(x),
+         map(to_symbolic, args(x)))
```

Comment:
This should be `Term{symtype(x)}(operation(x), map(to_symbolic, args(x)))` right?

Reply:
Well you're right! I'm about to test this :-p